modules/lighthouse_service.js

```diff
@@ -229,6 +229,87 @@ const failDestinationPlateBeckman = async (form) => {
   }
 }
 
+/**
+ * Format the plate specs to the expected type
+ * @param {*} plateSpecs a list of objects e.g. [{numberOfPlates: 1, numberOfPositives: 2}, {numberOfPlates: 3, numberOfPositives: 4}]
+ * @returns {list} plateSpecsList a string represtation of the plate specs in a nested list e.g. [[1,2],[3,4]]
+ */
+const formatPlateSpecs = (plateSpecs) => {
+  return plateSpecs.map((plate) => { return [plate.numberOfPlates, plate.numberOfPositives] })
+}
+
+// Create a test run
+const generateTestRun = async (plateSpecs, addToDart) => {
+  const plateSpecsParam = formatPlateSpecs(plateSpecs)
+  try {
+    const url = `${config.privateRuntimeConfig.lighthouseBaseURL}/cherrypick-test-data`
+    const body = {
+      'plate_specs': plateSpecsParam,
+      'add_to_dart': addToDart,
+    }
+    const headers = { headers: { Authorization: config.privateRuntimeConfig.lighthouseApiKey } }
+
+    const response = await axios.post(url, body, headers)
+
+    return {
+      success: true,
+      runId: response.data._id,
+    }
+  } catch (error) {
+    return {
+      success: false,
+      error: error.response ? error.response.data._error.message + ": " + JSON.stringify(error.response.data._issues) : 'An unexpected error has occured'
+    }
+  }
+}
+
+// Get all test runs
+const getTestRuns = async (currentPage, perPage) => {
+  try {
+    // TODO: refactor path
+    const url = `${config.privateRuntimeConfig.lighthouseBaseURL}/cherrypick-test-data?max_results=${perPage}&page=${currentPage}&sort=[("_created", -1)]`
+
+    const headers = { headers: { Authorization: config.privateRuntimeConfig.lighthouseApiKey } }
+
+    const response = await axios.get(url, headers)
+
+    // TODO: add total number of plates in run to response
+    // format barcodes from "[[\"TEST-112408\", \"number of positives: 0\"]]" here
+    return {
+      success: true,
+      response: response.data._items,
+      total: response.data._meta.total,
+    }
+  } catch (error) {
+    return {
+      success: false,
+      error: error.response ? error.response.data._error.message : 'An unexpected error has occured'
+    }
```

Comment:
Duplication with https://github.com/sanger/lighthouse-ui/pull/91/files#diff-3ca571c655124ab208a176ab716386bb5063b00ef54d7677eb93d6d5fb147cf2R305-R307

Reply:
If you mean compared to lines 305-308, sure.  You could extract a method that generates response dictionaries and takes a `success` parameter and a `details` parameter.  Depending on the `success` value, it would add either a `response` to the dictionary or an `error` to the dictionary.  If you also make it support the spread operator `...` you could pass it the `total` value on line 281 as well and it could insert that into the dictionary it creates by adding the `...` to the end of the dictionary when it's declared.

Reply:
Or, to be honest, just keep it simple and have two methods.  One for success and one for errors.  Add the logic for how to extract the error value inside the error method.

Reply:
Pulled out `errorResponse`